checker/src/main/java/org/checkerframework/checker/resourceleak/ResourceLeakAnnotatedTypeFactory.java

```diff
@@ -155,6 +155,35 @@ protected ResourceLeakAnalysis createFlowAnalysis() {
     return new ResourceLeakAnalysis((ResourceLeakChecker) checker, this);
   }
 
+  /**
+   * Retrieves the {@code @MustCall} annotation for the given object, which can be either an {@link
+   * Element} or a {@link Tree}. This method supports a polymorphic input type to facilitate easy
+   * retrieval of the {@code @MustCall} annotation from different types of objects within the
+   * type-checking framework.
+   *
+   * <p>This method delegates to the {@code MustCallAnnotatedTypeFactory} to get the annotated type
+   * of the input object and then extracts the primary {@code @MustCall} annotation from it. If the
+   * input object type is unsupported, an {@code IllegalArgumentException} is thrown.
```

Comment:
You don't need to say this, because it's redundant with the `@throws` clause below.